### ISSUE TITLE
Add support for Claude & Llama 3.x 70b with Groq

### ISSUE THREAD

**Issue by borisyankov** [callstackincubator/cali#25]
Add support and experiment how well these two models perform:

* Claude 3.5 Sonnet - often cited as much better than 4 (and even 4o) at coding
* Llama 3.x 70b  + Groq - not as good as the Claude or GPT 4o but inference is much faster and for many tasks it might be good enough

**Comment by grabbou** [comment]
Yup, ideally, we should allow configurability of the model, together with (if required) a model-specific prompt. The goal (and I think required next step) is to allow this as well as some self-hosted models.

I would also love to self-hosted something for everyone, so no API key is required. But it won't bring much to the table, unless I go bankrupt paying for it!

**Comment by adeleke5140** [comment]
Hi @grabbou, my team at [mastra](https://github.com/mastra-ai/mastra) has been building an AI framework for a while and we have support for these models out of the box.

We are also built on top of AI sdk so I believe it should be plug and play. We can onboard you if you're interested.

**Comment by grabbou** [comment]
Thanks for reaching out @adeleke5140 but we're not looking to adopt any higher-level framework at this point. At this point, we want to stay close to the metal and focus on experimenting with more tools instead.

I am not against this in general, especially as we grow, so let me keep this in the back of my head. Thanks!
